packages/wallet/src/redux/protocols/funding/player-a/reducer.ts

```diff
@@ -143,3 +173,38 @@ function cancelled(state: states.FundingState, sharedData: SharedData, action: a
       return unreachable(action.by);
   }
 }
+
+function handleFundingComplete(
+  protocolState: states.WaitForFunding,
+): states.WaitForSuccessConfirmation | states.Failure {
+  if (protocolState.fundingState.type === 'Success') {
+    return states.waitForSuccessConfirmation(protocolState);
+  } else {
+    // TODO: Indirect funding should return a proper error to pass to our failure state
+    return states.failure('Funding Failure');
+  }
+}
+function updateFundingState(
```

Comment:
It looks like there are two identical functions called `updateFundingState` introduced. Is there a reason for that?

Reply:
One for player A and one for player B. They are slightly different in that they are transitioning to the relevant player A or B states  at the end.

Reply:
This has been obsoleted regardless with the latest change.